tasks/December/tasks2/analyseSurvery/analyseSurvery.js

```diff
@@ -0,0 +1,94 @@
+// 4. Analiza wyników ankiety
+// Zadanie:
+// Napisz funkcję analyzeSurvey(results), która przyjmuje tablicę wyników
+// ankiety. Każdy wynik to obiekt z:
+// ● name (osoba),
+// ● answers (tablica odpowiedzi, gdzie każda odpowiedź ma question i
+// answer).
+// Funkcja powinna zwrócić:
+// 1. Liczbę unikalnych pytań.
+// 2. Tablicę osób, które odpowiedziały na wszystkie pytania.
+// 3. Obiekt z pytaniami jako kluczami i tablicą unikalnych odpowiedzi dla każdego
+// pytania.
+// Przykład danych wejściowych:
+const results = [
+  {
+    name: "Anna",
+    answers: [
+      { question: "Favorite color", answer: "Blue" },
+      { question: "Favorite food", answer: "Pizza" },
+    ],
+  },
+  {
+    name: "John",
+    answers: [
+      { question: "Favorite color", answer: "Green" },
+      { question: "Favorite food", answer: "Pizza" },
+    ],
+  },
+
+  {
+    name: "Kate",
+    answers: [{ question: "Favorite color", answer: "Blue" }],
+  },
+];
+
+// Oczekiwany wynik:
+// {
+// uniqueQuestions: 2,
+// allAnswered: ["Anna", "John"],
+// answersSummary: {
+// "Favorite color": ["Blue", "Green"],
+// "Favorite food": ["Pizza"]
+// }
+// }
+
+export const analyseSurvery = (results) => {
+  const getAllUniqueQuestions = () => {
+    const allQuestions = results.flatMap((singleResult) => {
+      return singleResult.answers.map((answer) => answer.question);
+    });
+
+    return Array.from(new Set(allQuestions));
+  };
+  const getNumberOfUniqueQuestions = () => {
+    return getAllUniqueQuestions().length;
+  };
+
+  const getUsernamesForAllQuestionsAnswered = () => {
+    const questions = getAllUniqueQuestions();
+
+    const usernames = results.filter((user) => {
```

Comment:
nie używałbym nazwy `user` na nazwe argumentu .filter(). Jest to mylące, ponieważ pod `user` kryje się cały obiekt nie tylko z nazwą usera a także z jego odpowiedzią. Nazwałbym to bardziej coś w stylu `result` albo `singleResult` etc